src/components/TeamMemberTasks/TeamMemberTasks.jsx

```diff
@@ -81,47 +88,33 @@ const TeamMemberTasks = props => {
   }, [currentUserId]);
 
   useEffect(() => {
-    renderTeamsList();
-  }, []);
-
-  useEffect(() => {
-    submitTasks();
-    if (userId !== props.auth.user.userid) {
-      dispatch(fetchTeamMembersTask(userId, props.auth.user.userid));
-      const currentUserRole = getUserRole(userId)
-        .then(resp => resp)
-        .then(user => {
-          setUserRole(user.data.role);
-        });
-    } else {
-      dispatch(fetchTeamMembersTask(userId, null));
-      setUserRole(props.auth.user.role);
+    if (isLoading === false && shouldRun) {
+      renderTeamsList();
+      closeMarkAsDone();
     }
-  }, [updatedTasks]);
+  }, [usersWithTasks, shouldRun]);
```

Comment:
Is this intended to run only once?
Is the expectation that when `isLoading` becomes `false`, that `usersWithTasks` will have updated?